File: backend/src/main/java/com/scinar/giderlerim/service/ButceService.java
package com.scinar.giderlerim.service;

import com.scinar.giderlerim.dto.request.ButceGuncelleRequest;
import com.scinar.giderlerim.dto.request.ButceOlusturRequest;
import com.scinar.giderlerim.dto.response.ApiResponse;
import com.scinar.giderlerim.dto.response.ButceOzetResponse;
import com.scinar.giderlerim.dto.response.ButceResponse;

import java.util.List;

public interface ButceService {
    ApiResponse<List<ButceResponse>> listele(Long kullaniciId, int ay, int yil);
    ApiResponse<ButceResponse> olustur(Long kullaniciId, ButceOlusturRequest request);
    ApiResponse<ButceResponse> guncelle(Long kullaniciId, Long butceId, ButceGuncelleRequest request);
    ApiResponse<Void> sil(Long kullaniciId, Long butceId);
    ApiResponse<List<ButceOzetResponse>> getAylikOzetler(Long kullaniciId, int ay, int yil);
}

File: packages/shared/services/butceService.ts
import { AxiosInstance } from 'axios';
import { ApiResponse } from '../types/api.types';
import { ButceOlusturRequest, ButceOzetResponse, ButceResponse } from '../types/butce.types';

export function createButceService(client: AxiosInstance) {
  return {
    async listele(ay: number, yil: number): Promise<ApiResponse<ButceResponse[]>> {
      const response = await client.get<ApiResponse<ButceResponse[]>>('/butceler', { params: { ay, yil } });
      return response.data;
    },

    async ozet(ay: number, yil: number): Promise<ApiResponse<ButceOzetResponse[]>> {
      const response = await client.get<ApiResponse<ButceOzetResponse[]>>('/butceler/ozet', { params: { ay, yil } });
      return response.data;
    },

    async olustur(data: ButceOlusturRequest): Promise<ApiResponse<ButceResponse>> {
      const response = await client.post<ApiResponse<ButceResponse>>('/butceler', data);
      return response.data;
    },

    async guncelle(id: number, data: { limitTutar?: number; uyariYuzdesi?: number }): Promise<ApiResponse<ButceResponse>> {
      const response = await client.put<ApiResponse<ButceResponse>>(`/butceler/${id}`, data);
      return response.data;
    },

    async sil(id: number): Promise<void> {
      await client.delete(`/butceler/${id}`);
    },
  };
}

export type ButceService = ReturnType<typeof createButceService>;
